scripts/changelog-header.sh

```diff
@@ -14,7 +14,7 @@ x.x.x Release notes (yyyy-MM-dd)
 * None.
 
 ### Compatibility
-* Realm Object Server: 3.21.0 or later.
+* Realm Object Server: 3.23.1 or later.
```

Comment:
@kneth I took the liberty to add this, so we don't have to remember to fix this at the next releases :-)